bson/types.go

```diff
@@ -39,47 +37,6 @@ const (
 	TypeMaxKey           = bsontype.MaxKey
 )
 
-var tBinary = reflect.TypeOf(primitive.Binary{})
```

Comment:
Looks like the reflection types defined here are also almost all defined in [`bsoncodec/types.go`](https://github.com/mongodb/mongo-go-driver/blob/master/bson/bsoncodec/types.go) (which is why so many are unused in this file). I wonder if we could just move the remaining `tRawValue` and `tRaw` types to `bsoncodec/types.go` to keep them all in one place.

Reply:
Moving those variables across packages would require exporting them, so I don't think that's a good idea. You're right that there's no much reason to keep these variables in `bson/types.go`, tho. `tRawValue` and `tRaw` are only used in `bson/primitive_codecs.go`, so I'll move them to that file. `tInt32` is only used in `bson/marshal_test.go` and `bson/unmarshal_test.go`, so I'll move that variable to one of those test files.

Reply:
Ah didn't think about the packages; moving `tRawValue`, `tRaw` and `tInt32` to those places sounds good to me.